nextjs/src/app/[locale]/c/[public_identifier]/CryptoWidgetPaymentButton.jsx

```diff
@@ -0,0 +1,357 @@
+"use client";
+// webpacker does not import the correct version automatically.
+// this is necessary for the Solana transfer object to function
+import * as buffer from "buffer";
+if (typeof window !== "undefined") {
+  window.Buffer = buffer.Buffer;
+}
+
+import { useContext } from "react";
+import { useTranslations } from "next-intl";
+import Web3 from "web3";
+import {
+  Connection,
+  SystemProgram,
+  LAMPORTS_PER_SOL,
+  Transaction,
+  PublicKey,
+} from "@solana/web3.js";
+import {
+  getAssociatedTokenAddress,
+  createAssociatedTokenAccountInstruction,
+  createTransferInstruction,
+} from "@solana/spl-token";
+
+import Button from "@brave/leo/react/button";
+import { CryptoWidgetContext } from "@/lib/context/CryptoWidgetContext";
+import styles from "@/styles/PublicChannelPage.module.css";
+import batAbi from "@/constant/batAbi.json";
+import erc20Abi from "@/constant/erc20Abi.json";
+
+export default function CryptoWidgetPaymentButton({
+  previewMode,
+  addresses,
+  cryptoConstants,
+}) {
+  const t = useTranslations();
+
+  const ethBatAddress = cryptoConstants.eth_bat_address;
+  const solanaBatAddress = cryptoConstants.solana_bat_address;
+  const solanaMainUrls = cryptoConstants.solana_main_urls;
+  const rpcHost = solanaMainUrls.filter((url) =>
+    url.toLowerCase().includes(window.location.host.toLowerCase()),
+  )[0];
+  const ethUsdcAddress = cryptoConstants.eth_usdc_address;
+  const solUsdcAddress = cryptoConstants.solana_usdc_address;
+
+  const {
+    currentChain,
+    currentAmount,
+    setIsSuccessView,
+    setErrorMsg,
+    setErrorTitle,
+    setIsTryBraveModalOpen,
+  } = useContext(CryptoWidgetContext);
+
+  async function sendPayment() {
+    clearError();
+    switch (currentChain) {
+      case "ETH":
```

Comment:
The switch statement is fine but you might consider storing this in a data structure as a mapping. e.g.) `{"ETH": sendEthPayment, "SOL": sendSolPayment, ...}`

Reply:
I like it!  Much shorter.